tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -385,6 +385,9 @@ def _fit(self, X_train, y_train, X_val=None, y_val=None, X_unlabeled=None, time_
         self.model = self._get_model()
 
         if X_unlabeled is not None:
+            # Can't spend all the time in pretraining, have to split it up.
+            if time_limit is not None:
+                time_limit = time_limit // 2
```

Comment:
why not just `time_limit / 2`?

Reply:
Also, if pretraining takes 2 minutes but 10 minutes was given as total time limit, would it make sense to use 8 minutes instead of 5 for the train fit?

Reply:
Can do time_limit / 2, at the time I was thinking it should be an int.

Reply:
> Also, if pretraining takes 2 minutes but 10 minutes was given as total time limit, would it make sense to use 8 minutes instead of 5 for the train fit?

Absolutely.

Reply:
New commit has the new time_limit logic in it.